mne/cuda.py

```diff
@@ -19,6 +21,48 @@
 _cuda_capable = False
 
 
+def _share_cuda_mem(x, n_jobs):
+    """Get shared memory space to avoid copying from cpu to gpu when possible.
+
+    Allocate a mapped ndarray with a buffer that is pinned and mapped on
+    to the device. Similar to np.empty()
+
+    Requires
+    --------
+    numba
+
+
+    Parameters
+    ----------
+    portable: bool
+        a boolean flag to allow the allocated device memory to be
+              usable in multiple devices.
+    wc: bool
+        a boolean flag to enable writecombined allocation which is faster
+        to write by the host and to read by the device, but slower to
+        write by the host and slower to write by the device.
+
+    Returns
+    -------
+    a mapped array: np.ndarray
+        An array to be passed into cupy.asarray, which does not copy if
+        shared memory is already allocated.
+    """
+    _soft_import("numba", "using shared memory")
+    from numba import cuda
+
```

Comment:
The import is properly nested below, don't need it here (and it's going to break filtering for anyone without cuda)
```suggestion
```

Reply:
copy ok.  yea I skipped this bc I didn't know where _soft_import came from, and didn't read my change from earlier.

Reply:
Yeah only need the `_soft_import` stuff if we're going to add the `get_shared_mem` to the public API, which I don't think we're going to do